lib/_http_agent.js

```diff
@@ -115,6 +115,9 @@ Agent.prototype.addRequest = function(req, options) {
     };
   }
 
+  options = util._extend({}, options);
```

Comment:
`Object.assign` is not available in v0.12 in case the change has to be merged there.